src/ssh.rs

```diff
@@ -42,16 +66,44 @@ impl Device for SshDevice {
         }
         Ok(())
     }
-    fn clean_app(&self, _exe: &path::Path) -> Result<()> {
-        unimplemented!()
+    fn clean_app(&self, app_path: &path::Path) -> Result<()> {
+        let user_at_host = format!("{}@{}", self.config.username, self.config.hostname);
+        let prefix = self.config.path.clone().unwrap_or("/tmp".into());
+        let app_name = app_path.file_name().unwrap();
+        let path = path::PathBuf::from(prefix)
+            .join("dinghy")
+            .join(app_name);
+        let stat = process::Command::new("ssh")
+            .arg(user_at_host)
+            .arg("-p")
+            .arg(&*format!("{}", self.config.port.unwrap_or(22)))
+            .arg(&*format!(
+                "rm -rf {}",
+                &path.to_str().unwrap()
+            ))
+            .status()?;
+        if !stat.success() {
+            Err("test fail.")?
+        }
+        Ok(())
     }
     fn run_app(&self, app_path: &path::Path, args: &[&str], envs: &[&str]) -> Result<()> {
         let user_at_host = format!("{}@{}", self.config.username, self.config.hostname);
+        let prefix = self.config.path.clone().unwrap_or("/tmp".into());
         let app_name = app_path.file_name().unwrap();
-        let path = path::PathBuf::from("/tmp/dinghy").join(app_name);
+        let path = path::PathBuf::from(prefix)
+            .join("dinghy")
+            .join(app_name);
         let exe = path.join(&app_name);
-        let stat = process::Command::new("ssh").arg(user_at_host)
-            .arg(&*format!("DINGHY=1 {} {}", envs.join(" "), &exe.to_str().unwrap()))
+        let stat = process::Command::new("ssh")
+            .arg(user_at_host)
+            .arg("-p")
+            .arg(&*format!("{}", self.config.port.unwrap_or(22)))
```

Comment:
this would break existing setups using the `~/.ssh/config` file to set the port, we should not add the `-p 22` arg if nothing is specified in dinghy conf